db/dynamodb.py

```diff
@@ -104,6 +108,38 @@ def __create_user_tables(self):
             }
         )
 
+    def __create_project_tables(self):
+        """
+        Create the project table.
+
+        **Note**: This function should **not** be called externally, and should
+        only be called on initialization, if at all.
+
+        Projects are only required to have a ``project_id``, technically. But we
+        also require that there be at least 1 URL in the ``github_urls`` field.
+        This is not programmically enforced.
+        """
+        logging.info("Creating table '{}'".format(self.projects_table))
+        self.ddb.create_table(
```

Comment:
As a side note, we might consider refactoring these into [CloudFormation](https://aws.amazon.com/cloudformation/) files - anything we request can be a simple JSON config file (that looks an awful lot like this request anyway) and then have a single "resource request" function that just reads the CloudFormation files and makes the request through CF.